### PR TITLE
Fix all PMD warnings throughout all modules

### LINKED ISSUE #184: Add PMD to build

### PR DESCRIPTION
**Pull request by amcp**
Fixes #184

### CODE REVIEW

File: ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/elassandra/ElassandraCassJavaDriverPlugin.java [hunk 1]
```diff
@@ -40,33 +43,48 @@
...
+    public void init(DataGenerator dataGenerator) {
         this.dataGenerator = dataGenerator;
+
+        this.ClusterName = configs.getCluster();
+        logger.info("Cassandra  Cluster: " + ClusterName);
+
+        this.ClusterContactPoint = configs.getHost();
+        this.KeyspaceName = configs.getKeyspace();
+        this.TableName = Optional.ofNullable(configs.getCfname()).orElse("external");
+        this.WriteConsistencyLevel = ConsistencyLevel.valueOf(Optional.ofNullable(configs.getWriteConsistencyLevel()).orElse(DEFAULT_CONSISTENCY_LEVEL));
```
Comment (amcp) [inline ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/elassandra/ElassandraCassJavaDriverPlugin.java]: pull into configuration subtype

**Comment by amcp** [comment]
@ipapapa ?

**Comment by vinaykumarchella** [comment]
Its in my list for this week

File: ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CJavaDriverBasePlugin.java [hunk 2]
```diff
@@ -7,111 +7,136 @@
...
-        ClusterContactPoint = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.host").asString("127.0.0.1").get();
-        KeyspaceName = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.keyspace").asString("dev1").get();
-        TableName =propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.cfname").asString("emp").get();
-        port = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE + "cass.host.port").asInteger(9042).get();
-        connections = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.connections").asInteger(2).get();
-
-        username = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.username").asString(null).get();
-        password = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.password").asString(null).get();
-
-        ReadConsistencyLevel = ConsistencyLevel.valueOf(propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.readConsistencyLevel").asString(ConsistencyLevel.LOCAL_ONE.toString()).get());
```
Comment (vinaykumarchella) [inline ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CJavaDriverBasePlugin.java]: Did you intentionally leave these out?

**Comment by vinaykumarchella** [comment]
@amcp LGTM except few nit comments

File: ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CassJavaDriverBatch.java [hunk 3]
```diff
@@ -59,15 +55,14 @@ void upsertCF(Session session) {
...
-
-    @Override
-    void postInit() {
-
+        this.TableName2 = config.getCfname2();
+        this.batchSize = config.getBatchSize();
+        this.useTimeStamp = config.getUseTimestamp();
+        this.useMultiPartition = config.getUseMultiPartition();
+        this.TableName = config.getCfname();
+        this.ReadConsistencyLevel = ConsistencyLevel.valueOf(config.getReadConsistencyLevel());
```
[the remark this answers is not in the document]
Comment (amcp) [reply ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CassJavaDriverBatch.java]: will pull back up

File: ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CJavaDriverBasePlugin.java (hunk 2 above)
Comment (amcp) [reply ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CJavaDriverBasePlugin.java]: will put back in

File: ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CassJavaDriverBatch.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (amcp) [reply ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CassJavaDriverBatch.java]: Actually, the Enum classes used to parse these strings are different for the plugins and the defaults might also be different. for example default MaxColCount is 100L in the base configuration but is 5L for Astynax.

File: ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CJavaDriverBasePlugin.java (hunk 2 above)
Comment (amcp) [reply ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CJavaDriverBasePlugin.java]: same answer is above, different Enum classes and different default values are used for Read/Write ConsistentcyLevel and MaxColCount. I will add a comment here and in implementing classes to indicate what is different.

**Comment by amcp** [comment]
@vinaykumarchella do you agree with the reasons why i left out a few configuration items from the base class?

**Pull request merged**